### PR TITLE
Only sync project cone when running source generators.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Our recent change to allow syncing project cones (without forking/dropping data on the oop side) means we can easily run SGs in oop while only having to wait for the current project cone to sync over.  This means a client can get the compilation for the current project (including generated documents) while only syncing that project and its depedencies.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker_Generators.cs [hunk 1]
```diff
@@ -121,6 +121,7 @@ private partial class CompilationTracker : ICompilationTracker
             var projectId = this.ProjectState.Id;
             var infosOpt = await connection.TryInvokeAsync(
                 compilationState,
+                projectId,
                 (service, solutionChecksum, cancellationToken) => service.GetSourceGenerationInfoAsync(solutionChecksum, projectId, cancellationToken),
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker_Generators.cs:125]: >(service, solutionChecksum, cancellationToken) [](http://example.com/codeflow?start=16&length=46)

Would it make sense for this callback to include the projectId?

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker_Generators.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker_Generators.cs:125]: will look into that :)

**Review by jasonmalinowski — COMMENTED** [review]
Ah, a nice small PR. :smile:

**Pull request merged**